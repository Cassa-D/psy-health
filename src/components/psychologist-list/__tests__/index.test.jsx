import { mount } from "enzyme";
import React from "react";
import { useSelector, useDispatch } from "react-redux";
import PsychologistList from "../index";
import StyledPsychologistList from "../styled";
import CardPsychologist from '../../card-psychologist'

jest.mock("react", () => ({
  // mock do useState
  ...jest.requireActual("react"),
}));

jest.mock("react-redux", () => ({
  // mock do useDispatch e do useSelector
  useDispatch: jest.fn(),
  useSelector: jest.fn(() => {}),
}));


describe("<PsychologistList />", () => {

 

  const wrapper = mount(<PsychologistList psychologists={[{ name: "joao", description:'teste apenas um test' }, { name: "pedro" , description:'teste apenas um test'}]} />);

  it("renders corectly", () => {
    expect(wrapper.find(StyledPsychologistList).children().find(CardPsychologist)).toHaveLength(2);
  });

  it(' snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('toBeTruthy', () => {
    expect(wrapper).toBeTruthy()
  })

});
