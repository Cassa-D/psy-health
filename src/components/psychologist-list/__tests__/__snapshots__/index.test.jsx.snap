// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PsychologistList />  snapshot 1`] = `ReactWrapper {}`;
